File: python/robot-name/robot_name.py
from string import ascii_uppercase as letters
from random import shuffle
from itertools import product

letter_pairs = (''.join(p) for p in product(letters, letters))
numbers = (str(i).zfill(3) for i in range(1000))
names = [l + n for l, n in product(letter_pairs, numbers)]
shuffle(names)
names = iter(names)


class Robot:
    def __init__(self):
        self.reset()

    def reset(self):
        self.name = next(names)

File: scala/robot-name/src/main/scala/RobotName.scala
import scala.util.Random

class Robot {
  var name: String = Robot.generateName()

  def reset() =
    name = Robot.generateName()
}

object Robot {
  def generateName(): String = {
    sample('A' to 'Z', 2) ++ sample('0' to '9', 3)
  } mkString


  def sample(range: Seq[Char], length: Int): Seq[Char] = {
    Seq.fill(length) {
      range(Random.nextInt(range.length))
    }
  }
}